ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_curvature_bases.m

```diff
@@ -35,10 +45,43 @@
             
             B_aug = (R\B_aug')';
             
+        case 'from-sys'
+		
+            % Import the datafile of the specified system
+            load(calcfilePath,'s');
             
+            % Calculate tolerance for find() (this will vary depending on
+                                            % system evaluation density)
+            tol=abs(s.grid.eval{1}(1,1)-s.grid.eval{1}(2,1))*0.5;
```

Comment:
I believe that what you're doing here is finding a closest grid point to the requested shape, then returning the transformation values at that grid point. Instead, I think you probably want to be using a smooth interpolating function to find the value of the transformation at the requested shape. This would mean you could replace the code from here through line 74 with:

r_cell = mat2cell(r);
x = interpn(s.grid.eval{:}, s.B_optimized.eval.Beta{1},r_cell{:}); % find the value of Bx for the given shape variables
y = interpn(s.grid.eval{:}, s.B_optimized.eval.Beta{2},r_cell{:}); % find the value of By for the given shape variables
theta = interpn(s.grid.eval{:}, s.B_optimized.eval.Beta{3},r_cell{:});% find the value of Btheta for the given shape variables

which also scales cleanly to systems with more than two shape variables.

Reply:
Done, thanks!